pkg/brownfield/targets.go

```diff
@@ -0,0 +1,145 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+package brownfield
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/to"
+
+	ptv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressprohibitedtarget/v1"
+)
+
+// Target uniquely identifies a subset of App Gateway configuration, which AGIC will manage or be prohibited from managing.
+type Target struct {
+	Hostname string
```

Comment:
when will we add `IP` field here ?

Reply:
Added a TODO.  Let's talk about implementing per-IP blacklists.  I think they'd be useful - but given the complexity this adds, I'm hesitant to implement that yet.  Gauging demand for this - we could implement in v1.1.